nameresolution/consul/consul_test.go

```diff
@@ -0,0 +1,754 @@
+package consul
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	nr "github.com/dapr/components-contrib/nameresolution"
+	"github.com/dapr/dapr/pkg/logger"
+	consul "github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockConsulResolver struct {
+	mock.Mock
```

Comment:
This is just my personal opinion. When possible I try to avoid using `github.com/stretchr/testify/mock` because _usually_ mocks can be accomplished more easily using Go interfaces. I created a Gist for you to see it applied in this file.

https://gist.github.com/pkedy/672d58762ccba9c2e100e9fa7ddf79a6